code/game/turfs/open.dm

```diff
@@ -71,7 +72,27 @@
 	if(scorchedness)
 		if(!icon_state_before_scorching)				//I hate you mappers, stop var editting turfs
 			icon_state_before_scorching = icon_state
-		icon_state = "[icon_state_before_scorching]_scorched[scorchedness]"
+		var/new_icon_state = "[icon_state_before_scorching]_scorched[scorchedness]"
+		if(icon_state != new_icon_state)				//no point in updating the icon_state if it would be updated to be the same thing that it was
+			icon_state = new_icon_state
+			for(var/i in GLOB.cardinals)				//but we still check so that we can update our neighbor's overlays if we do
+				var/turf/open/T = get_step(src, i)		//since otherwise they'd be stuck with overlays that were made with
+				T.update_icon()
+		for(var/i in GLOB.cardinals)
+			var/turf/open/T = get_step(src, i)
+			if(istype(T, /turf/open) && T.scorchable && T.scorchedness < scorchedness)
+				var/icon/edge_overlay
+				if(T.scorchedness)
+					edge_overlay = icon(T.icon, "[T.scorchable]_scorched[T.scorchedness]")
+				else
+					edge_overlay = icon(T.icon, T.scorchable)
+				if(!T.icon_state_before_scorching)
+					T.icon_state_before_scorching = T.icon_state
+				var/direction_from_nieghbor_towards_src = get_dir(T, src)
+				var/icon/culling_mask = icon(T.icon, "[T.scorchable]_mask[turf_edgeinfo_cache[T.icon_state_before_scorching][dir2indexnum(T.dir)][dir2indexnum(direction_from_nieghbor_towards_src)]]", direction_from_nieghbor_towards_src)
```

Comment:
```suggestion
				var/direction_from_neighbor_towards_src = get_dir(T, src)
				var/icon/culling_mask = icon(T.icon, "[T.scorchable]_mask[turf_edgeinfo_cache[T.icon_state_before_scorching][dir2indexnum(T.dir)][dir2indexnum(direction_from_nieghbor_towards_src)]]", direction_from_neighbor_towards_src)
```
minor nitpick gamer but otherwise lgtm